tests/D2L.CodeStyle.Analyzers.Test/Specs/ExternalDataSetVisibilityProviderAnalyzer.cs

```diff
@@ -0,0 +1,26 @@
+﻿// analyzer: D2L.CodeStyle.Analyzers.ApiUsage.DataHub.ExternalDataSetVisibilityProviderAnalyzer
+
+using System.Threading.Tasks;
+using System.Runtime.CompilerServices;
+
+namespace D2L.AW.DataExport.BrightspaceDataSets.Domain {
+	internal interface IEventDrivenDataSetPlugin { }
+	internal interface IExternalDataSetVisibilityProvider<T> where T : IEventDrivenDataSetPlugin { }
+}
+
+namespace D2L.CodeStyle.Analyzers.Tests.Specs {
+	using D2L.AW.DataExport.BrightspaceDataSets.Domain;
+
+	class ValidConstructorParameter : IEventDrivenDataSetPlugin {
+		ValidConstructorParameter(
+			IExternalDataSetVisibilityProvider<ValidConstructorParameter> e
+		) { }
+	}
+
+	class InvalidConstructorParameter : IEventDrivenDataSetPlugin{
+		InvalidConstructorParameter(
+			/* ExternalDataSetVisibilityProviderTypeParameterMatchesClass(IExternalDataSetVisibilityProvider, InvalidConstructorParameter) */ IExternalDataSetVisibilityProvider<ValidConstructorParameter> e /**/

```

Comment:
Two changes I made here:

- The assertion comment also asserts the message parameters, so they needed to be provided
- The comment needed to include the parameter name, since the diagnostic location for the whole parameter.